specification/eventgrid/data-plane/Microsoft.Web/stable/2018-01-01/Web.json

```diff
@@ -0,0 +1,573 @@
+      "properties": {
+        "action": {
+          "description": "Type of action of the operation.",
+          "type": "string",
+          "enum": [
+            "Restarted",
+            "Stopped",
+            "ChangedAppSettings",
+            "Started",
+            "Completed",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "AppAction",
+            "values": [
+              {
+                "value": "Restarted",
+                "description": "Web app was restarted."
+              },
+              {
+                "value": "Stopped",
+                "description": "Web app was stopped."
+              },
+              {
+                "value": "ChangedAppSettings",
+                "description": "There was an operation to change app setting on the web app."
+              },
+              {
+                "value": "Started",
+                "description": "The job has started."
+              },
+              {
+                "value": "Completed",
+                "description": "The job has completed."
+              },
+              {
+                "value": "Failed",
+                "description": "The job has failed to complete."
+              }
+            ]
+          }
+        }
+      }
+    },
+    "AppServicePlanEventTypeDetail": {
+      "description": "Detail of action on the app service plan.",
+      "type": "object",
+      "properties": {
+        "stampKind": {
+          "description": "Kind of environment where app service plan is.",
+          "type": "string",
+          "enum": [
+            "Public",
+            "AseV1",
+            "AseV2"
+          ],
+          "x-ms-enum": {
+            "name": "StampKind",
+            "values": [
+              {
+                "value": "Public",
+                "description": "App Service Plan is running on a public stamp."
+              },
+              {
+                "value": "AseV1",
+                "description": "App Service Plan is running on an App Service Environment V1."
+              },
+              {
+                "value": "AseV2",
+                "description": "App Service Plan is running on an App Service Environment V2."
+              }
+            ]
+          }
+        },
+        "action": {
+          "description": "Type of action on the app service plan.",
+          "type": "string",
+          "enum": [
+            "Updated"
+          ],
+          "x-ms-enum": {
+            "name": "AppServicePlanAction",
+            "values": [
+              {
+                "value": "Updated",
+                "description": "App Service plan is being updated."
+              }
+            ]
+          }
+        },
+        "status": {
+          "$ref": "#/definitions/AsyncStatus"
+        }
+      }
+    },
+    "AsyncStatus": {
+      "description": "Asynchronous operation status of the operation on the app service plan.",
+      "type": "string",
+      "enum": [
```

Comment:
Please add x-ms-enum, since most SDK won't treat the "enum" keyword alone and treat it as a string.